src/Illuminate/Database/Schema/Grammars/ChangeColumn.php

```diff
@@ -119,17 +128,35 @@ protected static function getDoctrineColumn(Table $table, Fluent $fluent)
      */
     protected static function getDoctrineColumnChangeOptions(Fluent $fluent)
     {
-        $options = ['type' => static::getDoctrineColumnType($fluent['type'])];
+        $doctrineType = static::getDoctrineColumnType($fluent['type']);
 
-        if (in_array($fluent['type'], ['text', 'mediumText', 'longText'])) {
+        $options = ['type' => $doctrineType];
+
+        if ($doctrineType->getName() === Types::TEXT) {
             $options['length'] = static::calculateDoctrineTextLength($fluent['type']);
         }
 
+        if ($fluent['type'] === 'binary') {
+            $options['length'] = AbstractMySQLPlatform::LENGTH_LIMIT_BLOB;
```

Comment:
Why use MySQL platform? What if user is using SQL Server?

Reply:
Laravel `binary` type maps to Doctrine `blob` type and only MySQL supports `length` option:

* MySQL: https://github.com/doctrine/dbal/blob/36b9ffc15c31598baa2a37b1abdd5b860d5c164f/src/Platforms/AbstractMySQLPlatform.php#L1270-L1289
* SQLServer: https://github.com/doctrine/dbal/blob/36b9ffc15c31598baa2a37b1abdd5b860d5c164f/src/Platforms/SQLServerPlatform.php#L1614-L1617
* PostreSQL: https://github.com/doctrine/dbal/blob/36b9ffc15c31598baa2a37b1abdd5b860d5c164f/src/Platforms/PostgreSQLPlatform.php#L1273-L1276
* SQLite: https://github.com/doctrine/dbal/blob/36b9ffc15c31598baa2a37b1abdd5b860d5c164f/src/Platforms/SqlitePlatform.php#L857-L860